pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,49 @@
+package declarativeconfig
+
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+type ClaimMapping struct {
+	PathToClaim    string `yaml:"pathToClaim,omitempty"`
+	MapToClaimName string `yaml:"mapTo,omitempty"`
```

Comment:
These a worded a bit unfortunate IMO, I think we can simplify them with proper documentation (i.e. comments here):
```yaml
claimMappings:
- path: some.claim.path
  name: new-name
```